### ISSUE TITLE
Set "Run as Admin"

### ISSUE THREAD

**Issue by SJRogue** [DmitriiShamrikov/mslinks#17]
Hi, Sorry to disturb you but I couldn't find it in doc or anything.
Does your API allow to set the Property "Run as Admin" for the shortcut?

Is this a cmd arg?

**Comment by SJRogue** [comment]
If not, does your API allow to change the target path?
I've tried ShellLink.setTarget() but check in Windows afterwards, it seems to not have an effect

**Comment by DmitriiShamrikov** [comment]
> Does your API allow to set the Property "Run as Admin" for the shortcut?

No, it doesn't. At least for now. As far as I remember there is no such flag in lnk file structure. Probably they use some trick for that option.

> I've tried ShellLink.setTarget() but check in Windows afterwards, it seems to not have an effect

It should work. Did you save the link file afterwards?

**Comment by tresf** [comment]
> As far as I remember there is no such flag in lnk file structure. Probably they use some trick for that option.

It's a registry entry.  It's only related to this project because the Windows Desktop illustrates it on the properties of the launcher.  A standard shell `reg` can add it without this library.

**Comment by Mike4Online** [comment]
On Windows 10, setting a desktop shortcut (for running an exe) to Run As Administrator alters the 22nd byte from 0x40 to 0x60.  Not sure if this holds for Windows Vista through Windows 8.1, or whether .lnk files  with this byte set to 0x60 would be OK on Windows XP.

**Comment by tresf** [comment]
@Mike4Online I have slightly different results.  Note, I'm using indices below for clarity.

Using `C:\ProgramData\Microsoft\Windows\Start Menu\Programs\Accessories\Paint.lnk`
* On Windows 10, I can confirm that `byte[21]` changes from `0x03` to `0x23` when the **Run As Administrator** flag is checked.
* On Windows XP, I can confirm that `byte[21]` changes from `0x02` to `0x22` when the **Run with different credentials** flag is checked.

Disclaimer: I'm using XVI32 to examine this with a standard hex editor.  If there's a byte offset that I should be using that may explain why our values differ.

Testing the Windows 10 byte changes on Windows XP SP3 show no immediate problems.
* If I force `0x03` value to match Windows 10, the shortcut still works as expected.
* If I force `0x23` value to match Windows 10, the shortcut prompts who to run the shortcut as (not identical in behavior, but expected, since Windows XP didn't have UAC, so this concept hadn't yet been invented.

(I'll update this post with Windows 7 results shortly)

> There is an API in Windows for setting the flags of a shortcut

Since this API works on a byte-level, that's the best approach.  There's ways to invoke the Windows APIs using a project called JNA, but that may be overkill given the simplicity of this using our combined research.

**Comment by tresf** [comment]
So @Mike4Online's proposal (or in my case, slightly varied propsoal) is tested and working on Windows XP, 7 and 10.  Without knowing the library very well, I would suggest making the API use bitwise OR value of `0x20`on `byte[21]`.  

@DmitriiShamrikov what are the next steps for someone such as @mike4online or myself to make a PR with this feature?  Reading the source, I assume there are some special considerations to be made and that the `.lnk` files are serialized using some specific guards or assumptions.  Can you point us in the right direction?
